src/com/content-comments/comments.js

```diff
@@ -165,26 +165,28 @@ export default class ContentComments extends Component {
 	}
 
 	renderPostNew() {
-		var user = this.props.user;
-		var authors = this.state.authors;
-		var comment = this.state.newcomment;
-		var author = authors[comment.author];
+		const user = this.props.user;
+		const authors = this.state.authors;
+		const comment = this.state.newcomment;
+		const author = authors[comment.author];
+		const allowAnonymous = parseInt(this.props.node.meta['allow-anonymous-comments']);
 
-		return <div class="-new-comment"><ContentCommentsComment user={user} comment={comment} author={author} indent={0} editing publish onpublish={this.onPublish} nolove /></div>;
+		return <div class="-new-comment"><ContentCommentsComment user={user} comment={comment} author={author} indent={0} editing publish onpublish={this.onPublish} nolove allowAnonymous={allowAnonymous} /></div>;
 	}
 
-	onPublish( e ) {
-		var node = this.props.node;
-		var newcomment = this.state.newcomment;
+	onPublish( e, publishAnon ) {
+		const node = this.props.node;
+		const newcomment = this.state.newcomment;
 
-		$Note.Add( newcomment.parent, newcomment.node, newcomment.body )
+		$Note.Add( newcomment.parent, newcomment.node, newcomment.body, null, publishAnon )
 		.then(r => {
 			if ( r.note ) {
 				var Now = new Date();
 				var comment = Object.assign({
-					'id': r.note,
-					'created': Now.toISOString(),
-					'modified': Now.toISOString()
+					id: r.note,
+					created: Now.toISOString(),
+					modified: Now.toISOString(),
+					anonymous: publishAnon,
 				}, newcomment);
```

Comment:
Please re-add the quotes around the member names, and around the new member (`'anonymous': ...`). This is for consistency with JSON.